### ISSUE TITLE
Should enums support min() and max()

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#21748]
Today, our enums are ordered and support comparisons like `<`, `>`, `<=`, etc. but do not support queries of their `min()` and `max()` values like other types.  Should they?  For example, given:

```chapel
enum color {red, green, blue};
writeln((min(color), max(color));
```

would write out:

```bash
(red, blue)
```

**Comment by mppf** [comment]
It would make sense to me for `min(color)` etc to work. Is there a use-case for this that motivates it?

**Comment by bradcray** [comment]
Yeah, I'm trying to write some generic code that wants to do min/max reductions on reasonably arbitrary-ish types, and one of the two cases(*) that fell down in my testing was enums.  More specifically, I'm parallelizing array.find() where we need to return the lexicographically smallest index (where, for enums, that would be the one earlier in the enumeration).  Applying a min reduction to the indices computed is a nice way of doing this, but requires a max value to store as the initial value.

(* = the other was strings, which seems challenging—what is a maximum value for strings?  Or should we just have a magic `maxString` built-in value that will always compare as being greater than any other string apart from itself?)

**Comment by vasslitvinov** [comment]
> Or should we just have a magic `maxString` built-in value that will always compare as being greater than any other string apart from itself?

I wonder if we can get away with maxString = a single character 0xff. It is invalid utf8 so any valid utf8 will be less than that. OTOH we can't play this trick with `bytes`, so maybe we need something else.
